homeassistant/components/sensor/bom.py

```diff
@@ -182,3 +217,69 @@ def update(self):
             _LOGGER.error("Check BOM %s", err.args)
             self.data = None
             raise
+
+
+def _get_bom_stations():
+    """Return {CONF_STATION: (lat, lon)} for all stations, for auto-config.
+
+    This function does several MB of internet requests, so please use the
+    caching version to minimise latency and hit-count.
+    """
+    latlon = {}
+    with io.BytesIO() as file_obj:
+        with ftplib.FTP('ftp.bom.gov.au') as ftp:
+            ftp.login()
+            ftp.cwd('anon2/home/ncc/metadata/sitelists')
+            ftp.retrbinary('RETR stations.zip', file_obj.write)
+        file_obj.seek(0)
+        with zipfile.ZipFile(file_obj) as zipped:
+            with zipped.open('stations.txt') as station_txt:
+                for _ in range(4):
+                    station_txt.readline()  # skip header
+                while True:
+                    line = station_txt.readline().decode().strip()
+                    if len(line) < 120:
+                        break  # end while loop, ignoring any footer text
+                    wmo, lat, lon = (line[a:b].strip() for a, b in
+                                     [(128, 134), (70, 78), (79, 88)])
+                    if wmo != '..':
+                        latlon[wmo] = (float(lat), float(lon))
+    zones = {}
+    pattern = (r'<a href="/products/(?P<zone>ID[DNQTVW]60801)/'
```

Comment:
Unfortunately not listed on the given pages when I checked, but I'll generalise this and the other check to accept any valid product ID instead of just those I found.